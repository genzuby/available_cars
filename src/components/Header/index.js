import React from 'react';
import { Nav } from 'react-bootstrap';
import { LOGOIMG, navStyle } from '../../style/componentStyle';

function Header() {
  return (
    <Nav className="container-fluid fixed-top" style={navStyle}>
      <Nav.Item>
        <Nav.Link className="d-flex" href="/home">
          <LOGOIMG
            className="align-self-center"
            src="/assets/logo.png"
            alt="Logo"
          />
        </Nav.Link>
      </Nav.Item>
    </Nav>
  );
}

export default Header;
