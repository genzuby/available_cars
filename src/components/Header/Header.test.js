import { render, screen, cleanup } from '@testing-library/react';
import Header from './index';

afterEach(cleanup);

beforeEach(() => {
  render(<Header />);
});

it('render snapshot', () => {
  expect(screen.getByTestId('header')).toMatchSnapshot();
});

it('shows bootstrap design properly', () => {
  expect(screen.getByTestId('header')).toHaveClass(
    'container-fluid fixed-top d-flex justify-content-between align-items-center'
  );
});

it('goes to Home link', () => {
  expect(screen.getByAltText('Logo').closest('a')).toHaveAttribute('href', '/');
});
