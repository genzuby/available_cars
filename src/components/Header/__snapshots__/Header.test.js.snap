// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render snapshot 1`] = `
<div
  class="container-fluid fixed-top d-flex justify-content-between align-items-center nav"
  data-testid="header"
  style="height: 80px; border-bottom: 2px solid #EDEDED; background-color: rgb(255, 255, 255); font-size: 20px;"
>
  <div
    class="nav-item"
  >
    <a
      class="nav-link"
      data-rb-event-key="/"
      href="/"
    >
      <img
        alt="Logo"
        class="sc-bdfBwQ leEZkR align-self-center"
        src="/assets/logo.png"
      />
    </a>
  </div>
  <div
    class="nav-item"
  >
    <a
      class="nav-link"
      data-rb-event-key="/favorites"
      href="/favorites"
    >
      <div
        class="sc-dlfnbm iWuJww"
        color="#4A4A4A"
        font-size="18px"
      >
        My Favorites
      </div>
    </a>
  </div>
</div>
`;
